### PR TITLE
Added tassign, docs about typing reducers

### LINKED ISSUE #216: Documentation idea: advice about strongly typing reducers.
Lots of great commentary on this topic as part of the discussion of issue #210 . Let's get this stuff captured in an 'in-depth usage' article in the docs.

### PR DESCRIPTION
**Pull request by SethDavenport**
Fixes #216

### CODE REVIEW

**Comment by SethDavenport** [comment]
Ping :)

File: docs/strongly-typed-reducers.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+export const barReducer = (state: IBar, action): IBar => {
+  // ...
+};
+```
+
+Since most applications are composed of several reducers, you should compose 
+a global 'AppState' by composing the reducer types:
+
+```typescript
+export interface IAppState {
```
Comment (e-schultz) [inline docs/strongly-typed-reducers.md]: What about:
```js
export interface IAppState {
   foo?: TFoo;
   bar?: IBar;
 }
```

So you're able to provide an empty object when configuring the store. I pretty much do it this way all the time.

File: src/utils/tassign.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export function tassign<T extends U, U>(target: T, ...source: U[]): T {
```
Comment (e-schultz) [inline src/utils/tassign.ts]: Thoughts on making `tassign` it's own NPM package? can see this being useful outside of `ng2-redux`.

File: docs/strongly-typed-reducers.md (hunk 1 above)
Comment (SethDavenport) [inline docs/strongly-typed-reducers.md]: Yes good point.

File: src/utils/tassign.ts (hunk 2 above)
Comment (SethDavenport) [inline src/utils/tassign.ts]: Makes sense/

**Comment by SethDavenport** [comment]
@bennett000 just pointed out Typescript 2's `readonly`. Seems very useful... I want to play with it and see if it can improve things even more.

File: src/utils/tassign.ts (hunk 2 above)
Comment (SethDavenport) [inline src/utils/tassign.ts]: see angular-redux/tassign

**Pull request merged**